### PR TITLE
update deployment in k8s and add available time for notification,optimize create address.

### PR DESCRIPTION
**Pull request by huojiao2006**

### CODE REVIEW

File: pkg/gerr/message.go [hunk 1]
```diff
@@ -118,4 +118,14 @@ var (
 		en:   "address_list [%s] not exist",
 		zhCN: "地址列表[%s]不存在",
 	}
+	ErrorIllegalTimeFormat = ErrorMessage{
+		Name: "illegal_time_format",
+		en:   "illegal time format [%s]",
+		zhCN: "错误的时间格式[%s]",
+	}
+	ErrorNotAvailableTimeAvailable = ErrorMessage{
```
Comment (chilianyi) [inline pkg/gerr/message.go]: ErrorNotAvailableTimeRange?

File: pkg/plugins/email_notifier.go [hunk 2]
```diff
@@ -6,9 +6,10 @@ package plugins
 
 import (
 	"context"
-
+	"openpitrix.io/logger"
```
Comment (chilianyi) [inline pkg/plugins/email_notifier.go:10]: need empty line.

File: pkg/util/stringutil/string.go [hunk 3]
```diff
@@ -5,8 +5,10 @@
 package stringutil
 
 import (
+	"openpitrix.io/notification/pkg/gerr"
```
Comment (chilianyi) [inline pkg/util/stringutil/string.go]: Should not use "openpitrix.io/notification/pkg/gerr" in util package.
May cause loop rely.

File: deploy/db/mysql.yaml [hunk 4]
```diff
@@ -73,7 +73,7 @@ spec:
         - name: MYSQL_ROOT_PASSWORD
           valueFrom:
             secretKeyRef:
-              key: password.txt
+              key: password
```
Comment (chilianyi) [inline deploy/db/mysql.yaml:76]: Where is the password file?

Comment (huojiao2006) [reply deploy/db/mysql.yaml:76]: now just set the value, not use a password file.

**Review by chilianyi — APPROVED** [review]

File: pkg/util/stringutil/string.go [hunk 5]
```diff
@@ -84,3 +85,29 @@ func SimplifyString(s string) string {
 }
 
 var reMoreSpace = regexp.MustCompile(`\s+`)
+
+func CheckTimeAvailable2(availableStartTime time.Time, availableEndTime time.Time) bool {
```
Comment (chilianyi) [inline pkg/util/stringutil/string.go]: useful？

File: deploy/db/mysql.yaml [hunk 6]
```diff
@@ -30,7 +30,7 @@ spec:
     - ReadWriteOnce
   resources:
     requests:
-      storage: 20Gi
+      storage: 5Gi
```
Comment (chilianyi) [inline deploy/db/mysql.yaml]: Screenshot from 2019-03-26 11-31-17
Better larger than 10G

File: pkg/db/schema/notification/V0_3__add_available_time.sql [hunk 7]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE notification
+	ADD COLUMN available_start_time varchar(8) default '';
+
+
+ALTER TABLE notification
+	ADD COLUMN available_end_time   varchar(8) default  '';
```
Comment (chilianyi) [inline pkg/db/schema/notification/V0_3__add_available_time.sql]: need fmt

File: pkg/gerr/message.go [hunk 8]
```diff
@@ -118,4 +118,14 @@ var (
...
 		zhCN: "地址列表[%s]不存在",
 	}
+	ErrorIllegalTimeFormat = ErrorMessage{
+		Name: "illegal_time_format",
+		en:   "illegal time format [%s]",
+		zhCN: "错误的时间格式[%s]",
+	}
+	ErrorNotAvailableTimeRange = ErrorMessage{
+		Name: "not_available_time",
+		en:   "not available time [%s]-[%s",
```
Comment (chilianyi) [inline pkg/gerr/message.go]: lack ] at the end.

File: pkg/util/stringutil/string.go [hunk 9]
```diff
@@ -84,3 +85,29 @@ func SimplifyString(s string) string {
...
+}
+
+func CheckTimeAvailable(availableStartTimeStr string, availableEndTimeStr string) bool {
+	timeFmt := "15:04:05"
+	currentTime := time.Now().Format(timeFmt)
+	currentTime1, _ := time.Parse(timeFmt, currentTime)
+
+	availableStartTime, _ := time.Parse(timeFmt, availableStartTimeStr)
+	availableEndTime, _ := time.Parse(timeFmt, availableEndTimeStr)
+	if availableStartTime.Before(currentTime1) && availableEndTime.After(currentTime1) {
```
Comment (zheng1) [inline pkg/util/stringutil/string.go]:
```
return availableStartTime.Before(currentTime1) && availableEndTime.After(currentTime1)
```

File: pkg/services/notification/handler_test.go [hunk 10]
```diff
@@ -485,3 +507,20 @@ func TestValidateEmailService(t *testing.T) {
...
+
+func TestCheckTimeAvailable(t *testing.T) {
+	if !*pkg.LocalDevEnvEnabled {
+		t.Skip("Local Dev testing env disabled.")
+	}
+	logger.SetLevelByString("debug")
+
+	availableStartTimeStr := "00:00:00"
+	availableEndTimeStr := "20:00:00"
+	isOK := stringutil.CheckTimeAvailable(availableStartTimeStr, availableEndTimeStr)
```
Comment (zheng1) [inline pkg/services/notification/handler_test.go]: cannot pass this test at 21:00:00

**Pull request merged**